From ba5f9fe47c586806cb45172c760ce9aeeb85af50 Mon Sep 17 00:00:00 2001
From: Michal Simek <monstr@monstr.eu>
Date: Tue, 3 Apr 2012 12:57:12 +0200
Subject: [PATCH] arm: zynq: Move compiled-in dtb behind .text section

Moving dtb behind .text section fix problem with
loadable modules because it was placed in init section
which is cleared. That's why dtb was corrupted and
drivers weren't probed.

Signed-off-by: Michal Simek <monstr@monstr.eu>
---
 arch/arm/kernel/vmlinux.lds.S |   23 +++++++++++++----------
 1 files changed, 13 insertions(+), 10 deletions(-)

diff --git a/arch/arm/kernel/vmlinux.lds.S b/arch/arm/kernel/vmlinux.lds.S
index 0dbc814..28229bb 100644
--- a/arch/arm/kernel/vmlinux.lds.S
+++ b/arch/arm/kernel/vmlinux.lds.S
@@ -116,6 +116,19 @@ SECTIONS
 
 	RO_DATA(PAGE_SIZE)
 
+
+#ifdef CONFIG_ARCH_ZYNQ
+	. = ALIGN (4) ;
+	__fdt_blob : AT(ADDR(__fdt_blob) - LOAD_OFFSET) {
+		_fdt_start = . ;		/* place for fdt blob */
+		*(__fdt_blob) ;			/* Any link-placed DTB */
+	        . = _fdt_start + 0x8000;	/* Pad up to 16kbyte */
+		_fdt_end = . ;
+	}
+#endif
+
+
+
 #ifdef CONFIG_ARM_UNWIND
 	/*
 	 * Stack unwinding tables
@@ -187,16 +200,6 @@ SECTIONS
 
 	PERCPU_SECTION(L1_CACHE_BYTES)
 
-#ifdef CONFIG_ARCH_ZYNQ
-	. = ALIGN (4) ;
-	__fdt_blob : AT(ADDR(__fdt_blob) - LOAD_OFFSET) {
-		_fdt_start = . ;		/* place for fdt blob */
-		*(__fdt_blob) ;			/* Any link-placed DTB */
-	        . = _fdt_start + 0x8000;	/* Pad up to 16kbyte */
-		_fdt_end = . ;
-	}
-#endif
-
 #ifdef CONFIG_XIP_KERNEL
 	__data_loc = ALIGN(4);		/* location in binary */
 	. = PAGE_OFFSET + TEXT_OFFSET;
-- 
1.7.0.4

